src/librustc/hir/lowering.rs

```diff
@@ -2969,7 +2969,7 @@ impl<'a> LoweringContext<'a> {
                     let body = if let Some(else_expr) = wildcard_arm {
                         P(self.lower_expr(else_expr))
                     } else {
-                        self.expr_tuple(e.span, hir_vec![])
+                        self.expr_tuple(body.span, hir_vec![])
```

Comment:
Why the change to body.span here? It's not entirely intuitive to me that the (synthetic) "else clause" should get the span of the code that executes when `else` does not apply. =) At minimum, a comment seems good.

Reply:
Reverted.